agents/lakeshore336/LS336_agent.py

```diff
@@ -0,0 +1,782 @@
+
+        Parameters
+        ----------
+        params : dict
+            Contains parameters 'threshold' (optiona, default 0.1), 'window' (option, default 900),
+            'heater' (optional, default '2'), and 'wait' (optional, default 1).
+
+        Notes
+        -----
+        Param 'threshold' sets the upper bound on the absolute temperature difference between
+        the setpoint and any temperature from the input channel in the last 'window' seconds.
+
+        Param 'window' sets the lookback time into the most recent temperature data, in seconds.
+        '''
+
+        # get threshold
+        threshold = params.get('threshold')
+        if threshold is None: threshold = self.threshold
+
+        # get window
+        window = params.get('window')
+        if window is None: window = self.window
+        num_idxs = int(np.ceil(window / self.t_sample))
+
+        with self.lock.acquire_timeout(job = 'check_temperature_stability', timeout = 3) as acquired:
+            if not acquired:
+                print(f"Lock could not be acquired because it is held by {self.lock.job}")
+                return False, 'Could not acquire lock'
+
+            session.set_status('running')
+
+            # get heater
+            heater_key = params.get('heater', '2') # default to 50W output
+            heater = self.module.heaters[heater_key]
+
+            # get channel
+            channel = heater.get_input_channel()
+            channel_num = self.module.channels[channel].num
+
+            # get current temp
+            current_temp = np.round(float(self.module.get_kelvin(channel)), 4)
+
+            # check if recent temps and current temps are within threshold
+            recent_temps = self.recent_temps[:num_idxs, channel_num-1]
+            recent_temps = np.concatenate((np.array([current_temp]), recent_temps))
+
+            # get static setpoint if None
+            if self.static_setpoint is None:
+                self.static_setpoint = float(heater.get_setpoint())
+
+            setpoint = self.static_setpoint # avoids ramp bug, i.e. want to compare to commanded setpoint not mid-ramp setpoint
+            session.add_message(f'Maximum absolute difference in recent temps is {np.max(np.abs(recent_temps - setpoint))}K')
+            
+            if np.all(np.abs(recent_temps - setpoint) < threshold):
+                session.add_message(f'Recent temps are within {threshold}K of setpoint')
+                return True, f'Servo temperature is stable within {threshold}K of setpoint'
+
+            else:
+                print('Servo temps are not yet stable')
+
+            time.sleep(params.get('wait', self.wait))
+        
+        return False, f'Servo temperature is not stable within {threshold}K of setpoint'
+            
+    def get_channel_attribute(self, session, params):
+        '''Gets an arbitrary channel attribute, stored in the session.data dict
```

Comment:
I'd reword this to something like "Gets an arbitrary channel attribute and stores it in the `session.data` dict." Right now it sounds like your getting the attribute from the `session.data` dict.

Please also add an example for the structure of the data stored within `session.data` for reference in the documentation. See the 372 Agent acquisition process for an example: https://github.com/simonsobs/socs/blob/b6dde01c3f66dc4dcc528262de7428ba19d88b95/agents/lakeshore372/LS372_agent.py#L182

Reply:
1. done
2. done